core/federated/RTI/rti_common.h

```diff
@@ -246,25 +257,69 @@ tag_t earliest_future_incoming_message_tag(scheduling_node_t* e);
 tag_t eimt_strict(scheduling_node_t* e);
 
 /**
- * Return true if the node is in a zero-delay cycle.
+ * For the given scheduling node (enclave or federate), if necessary, update the `min_delays`,
+ * `all_upstremas`, `num_all_upstreams`, and the fields that indicate cycles.  These fields will be
+ * updated only if they have not been previously updated or if invalidate_min_delays_upstream
+ * has been called since they were last updated.
  * @param node The node.
  */
-bool is_in_zero_delay_cycle(scheduling_node_t* node);
+void update_min_delays_upstream(scheduling_node_t* node);
 
 /**
- * Return true if the node is in a cycle (possibly a zero-delay cycle).
+ * For the given scheduling node (enclave or federate), if necessary, update the `all_downstreams`,
+ * `num_all_downstreams`.  These fields will be updated only if they have not been previously updated
+ * or if invalidate_min_delays_upstream has been called since they were last updated.
```

Comment:
```suggestion
 * For the given scheduling node (enclave or federate), if necessary, update the `all_downstreams` and
 * `num_all_downstreams` fields.  These fields will be updated only if they have not been previously updated
 * or if invalidate_min_delays_upstream has been called since they were last updated.
```
This is a little confusing.  Should `invalidate_min_delays_upstream` be renamed `invalidate_min_delays`?  From this comment, it seems it also invalidates the downstream delays.

Reply:
You're right. Thank you for pointing out this! I'll work on it along with the related comments you provided.

Reply:
Resolved in [4507043](https://github.com/lf-lang/reactor-c/pull/349/commits/4507043ca782ed8e19901fc58fd5cf5ea06422d9).